internal/app/machined/pkg/controllers/siderolink/manager.go

```diff
@@ -246,6 +246,13 @@ func (ctrl *ManagerController) Run(ctx context.Context, r controller.Runtime, lo
 		linkSpec := network.NewLinkSpec(network.ConfigNamespaceName, network.LayeredID(network.ConfigOperator, network.LinkID(constants.SideroLinkName)))
 		addressSpec := network.NewAddressSpec(network.ConfigNamespaceName, network.LayeredID(network.ConfigOperator, network.AddressID(constants.SideroLinkName, nodeAddress)))
 
+		endpoints := resp.GetEndpoints()
```

Comment:
this is not that easy... we should support cycling through the endpoints if the connection is down (today we just do full reconnect, but we should try all endpoints first)

Reply:
Hmm, I tried approaching this by storing the endpoints response in the controller and rotating through them until we reach the 0 idx, in which case we try the provisioning again.